servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -194,7 +194,11 @@ private void handleMasterCommand(Command cmd) throws IOException {
       // Master requests blocks to be removed from Tachyon managed space.
       case Free:
         for (long block : cmd.mData) {
-          mBlockDataManager.removeBlock(Users.MASTER_COMMAND_USER_ID, block);
+          try {
+            mBlockDataManager.removeBlock(Users.MASTER_COMMAND_USER_ID, block);
+          } catch (IOException ioe) {
+            LOG.warn("Failed master delete block cmd for: " + block + " due to concurrent read.");
```

Comment:
"delete" sounds like the block is being deleted forever. Maybe this should be called free or remove or something?